### PR TITLE
Remove some of the magic from run-test.sh

### PR DESCRIPTION
**Pull request by ellismg**
Run-Test did a bunch of work to try to fix up the tests folder produced by a windows build so that it could run on Linux or OSX.  With #3676 much of that work is no longer needed and can be deleted.  This gets us closer to having the cross platform test runs behave more like the windows ones.

### CODE REVIEW

**Comment by ellismg** [comment]
@weshaggard @stephentoub can you please take a look?

/cc @jasonwilliams200OK @janhenke  I think this a net improvement for the FreeBSD effort, but I can't be sure.  I'm sorry if this sets you back and we can work to get you unblocked.

My suspicion is this helps you.  I think what you guys will want to do is use ubuntu-14.04 as the RID for FreeBSD in build.sh (until we have a real FreeBSD RID).  This will cause the "correct" managed components to be restored but the native components will be wrong.  However, run-test.sh can copy over the native components that you built on FreeBSD.

This gets rid of the need for that temporary host package which was going to be busted for you anyway.

We need to merge this in even if it sets you back a bit because we are in a very bad place now with our CI system (it was on the floor this morning due to the fact that the managed copy in run-test.sh is super fragile and can break depending on how `find` enumerates the files in the binaries folder).  Let me know if you have additional concerns and we can figure out how to address them after this goes in.

/cc @mmitche as an FYI.

**Comment by stephentoub** [comment]
LGTM

**Comment by ellismg** [comment]
@pgavlin With these changes I expect your workflow on FC 22 can move to just building CoreCLR and CoreFX on Linux and then using run-test.sh, passing the path to the CoreCLR and MSCorlib you built.

The defaults for --corefx-tests should be correct now and you shouldn't need to pass it explicitly.

If this doesn't work for you, let me know and I can work on fixing it.

You will still need to run tests using run-test.sh instead of from MSBuild directly because you need the special copy behavior for the runtime and native components.

**Comment by mmitche** [comment]
@ellismg This is fantastic, great to see those temporary test packages go away.

**Comment by weshaggard** [comment]
Yes this is great. Thanks for doing this @ellismg

File: run-test.sh [hunk 1]
```diff
@@ -214,6 +172,8 @@ runtest()
   then
     rm mscorlib.ni.dll
   fi
+
+  chmod +x ./corerun
```
Comment (pgavlin) [inline run-test.sh]: Just out of curiosity, why is this necessary? On which platformsis `corerun` not already executable?

Comment (ellismg) [inline run-test.sh]: I was only seeing this in CI.  My guess is that when we copy the CoreCLR build (which contains corerun) from the Jenkins server, the execute bit is not preserved.

I think that long term this will also be needed, since when you restore corerun from a nuget package, it will not be executable.

**Comment by jasonwilliams200OK** [comment]
@ellismg, somehow i missed this PR..

Yes it seems like an extra work but IMHO, [before I was copying stuff twice](https://github.com/dotnet/coreclr/issues/1633#issuecomment-143669303): CoreCLR bins and mscorlib separately to be able to execute `run-test.sh`. This change actually reduces the number of cp to one. Thanks! :tada: 

I am building CoreFX on (again) new FreeBSD VM to test these changes (it takes me 18 minutes on Windows 10 on average to get the new [FreeBSD 10.2 VM ready with everything](https://gist.github.com/jasonwilliams200OK/6efa7907e66275df2d24) to execute `~/corefx/build.sh` :smile:). If I get time, I am planning to create CentOS, Ubuntu and FreeBSD `Dockerfile`s for Core[FX/CLR] and Roslyn to get the logs and binaries out of it. Besides, linking against libc/libc++ provided by old CentOS-5 will make the native binaries "vastly" compatible with all Linices (otherwise if we distribute binaries built on Ubuntu 14, it will not work with old CentOS (especially v5x, which many hosting services are running) and old Fedoras, unless we static compile libc[xx] and get our bins bulky).

Will get back with more updates once the build finishes and there is something new about CoreFX test experience on FreeBSD. Cheers! :beers:

File: run-test.sh (hunk 1 above)
Comment (jasonwilliams200OK) [inline run-test.sh]: I have the same curiosity. :D

**Comment by jasonwilliams200OK** [comment]
@ellismg, incidentally, does this mean we need to add `__TestNugetRuntimeId=ubuntu.14.04-x64` after https://github.com/dotnet/corefx/blob/830fe6b883c9e5a17a9ef9767621a468ca73f9c4/build.sh#L302 everytime we execute `./build.sh` on FreeBSD? At least this is what I did to get to [this point](https://github.com/dotnet/corefx/issues/1626#issuecomment-146330969) (40 tests failing).

BTW, BSD OSes are more OSX like (or the opposite?) than Linux. :smile:

**Comment by ellismg** [comment]
Yes, it does.  You could probably use the OSX rid if you wanted.  In general the mangled code will be the same between the two and the native code will be wrong for both.  With the shims work we are trying to move away from platform specific managed code, and the native shims are tied heavily to the target OS.
